src/data_validation/validation.py

```diff
@@ -0,0 +1,110 @@
+import postcodes_uk
+import pandas as pd
+
+from src.utils.wrappers import logger_creator
+from src.utils.helpers import Config_settings
+
+
+# Get the config
+conf_obj = Config_settings()
+config = conf_obj.config_dict
+global_config = config["global"]
+
+# Set up logging
+logger = logger_creator(global_config)
+
+
+def validate_postcode_pattern(pcode: str) -> bool:
+    """A function to validate UK postcodes which uses the
+
+    Args:
+        pcode (str): The postcode to validate
+
+    Returns:
+        bool: True or False depending on if it is valid or not
+    """
+    if pcode is None:
+        return False
+
+    # Validation step
+    valid_bool = postcodes_uk.validate(pcode)
+
+    return valid_bool
+
+
+def get_masterlist(masterlist_path) -> pd.Series:
+    """This function loads the masterlist of postcodes from a csv file
+
+    Returns:
+        pd.Series: The dataframe of postcodes
+    """
+    masterlist = pd.read_csv(masterlist_path, usecols=["pcd"]).squeeze()
```

Comment:
Changed this since it was failing. Squeeze function converts the pcd column to pd.series.

Reply:
Nice